### PR TITLE
Improve file format detection

### PR DESCRIPTION
**Pull request by lassoan**
Improves automatic detection of segmentation and NIFTI transform files.

### CODE REVIEW

File: Modules/Loadable/Transforms/qSlicerTransformsReader.h [hunk 1]
```diff
@@ -47,6 +47,13 @@ class qSlicerTransformsReader
 
   bool load(const IOProperties& properties) override;
 
+  /// Returns a positive number (>0) if the reader can load this file.
+  /// It uses default confidence value except for NIFTI files.
+  /// For NIFTI file that contain displacement field, the method returns
+  /// higher than default (0.6) to make the application prefer reading it as transform.
```
Comment (jcfr) [inline Modules/Loadable/Transforms/qSlicerTransformsReader.h:53]:
```suggestion
  /// For NIFTI file that contain displacement field, the method returns 0.6 which is
  /// higher than default (0.5) to make the application prefer reading it as transform.
```

**Review by pieper — APPROVED** [review]
This looks great 👍 

I don't think it's needed for this, but someday if we really care about covering nifti more 'correct' uses of nifti we should assemble a set of 'correct' example files along with common variants that come up in practice.  But for now just dealing with cases as they come up seems good enough.

Did you want to include a link to the discourse thread about SynthMorph which I guess inspired this?

**Comment by lassoan** [comment]
Thanks for the reviews. I've accepted Jcfr's change and added link and description of NIFTI displacement field format as Steve suggested.

**Review by pieper — APPROVED** [review]
LGTM 👍

**Pull request merged**